phac-nml/irida-next
# Public activity: Projects

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in an `Activity` page to a **project** which will list the following activities:

| Type  | Action |
| ------------- | ------------- |
| Project  | Created, Updated  |
| Member  | Added, Updated, Removed, Leaves Project  |
| Shared Group  | Shared, Unshared  |
| Sample  | Added, Removed, Updated, Transferred, Cloned  |
| Sample Attachments  | Attached, Removed  |
| Sample Metadata  | Updated (Added, removed, changed)  |
| Automated Workflow Executions  | Created, Launched, Write outputs/metadata to samples  |

Note: Sample and group level activity will come in followup PRs

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/user-attachments/assets/533344bb-d95a-4ed6-8d97-fcd0160ce150)

![image](https://github.com/user-attachments/assets/0f8d756b-dab2-44f2-a340-56b8c5931083)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Create 2 projects, `ProjA` and `ProjB`
2. In `ProjA` create 3 samples
3. Visit the project `Activity` page from the project `Settings` menu
4. You should see activity entries for the created project and 3 entries for created samples
5. Add a member to the project from the `Members` page. Now remove this member
6. From the activity page verify that it has 2 new entries, one for adding a user and another for removing
7. Setup an `Automated Workflow Execution` for the project from the `Settings` menu and select the option to `Update samples with analysis results`
8. From the activity page verify an entry for the user setting up an automated workflow execution for the project
9. Go into the samples for the project and upload paired-end data to one of the samples
10. From the activity page verify a new entries one for adding sequencing data to a sample and one for the project automation launching a workflow
11. When the workflow has completed check the activity page for the project and verify an entry for workflow outputs and metadata written to the sample which was used for the workflow execution
12. Remove one of the samples from the project and on the activity page verify a new entry for a sample being removed
13. Clone a sample to `ProjB` and then also transfer a sample to `ProjB`
14. From the `ProjA` activity page see 2 new entries, one for cloning samples to `ProjB` and the other for transferring samples to `ProjB`
15. From the `ProjB` activity page verify the 2 newest entries, one for cloning samples from `ProjA` and the other for transferring samples from `ProjA`
16. From the samples page for `ProjB`, click on one of the samples, then go to the metadata tab
17. Add some metadata to the sample
18. From the activity page for `ProjB` verify an entry is visible for updated metadata for a sample

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
